ext/standard/tests/strings/locale_independent_float_to_string.phpt

```diff
@@ -0,0 +1,86 @@
+--TEST--
+Test that floats are converted to string locale independently
+--SKIPIF--
+<?php
+if (!setlocale
+    (LC_ALL,
+    "german", "de", "de_DE", "de_DE.ISO8859-1", "de_DE.ISO_8859-1", "de_DE.UTF-8",
+    "french", "fr", "fr_FR", "fr_FR.ISO8859-1", "fr_FR.ISO_8859-1", "fr_FR.UTF-8",
+    )) {
+    die("skip locale needed for this test is not supported on this platform");
+}
+?>
+--FILE--
+<?php
+
+function print_float(float $f)
+{
+    echo "- casting:\n";
+    echo $f . "\n";
+    echo strval($f) . "\n";
+
+    echo "- *printf functions:\n";
+    printf("%.2f\n", $f);
+    printf("%.2F\n", $f);
+    echo sprintf("%.2f", $f) . "\n";
+    echo sprintf("%.2F", $f) . "\n";
+
+    echo "- exporting:\n";
+    echo var_export($f, true)  . "\n";
+    echo serialize($f) . "\n";
+    echo json_encode($f) . "\n";
+
+    echo "- debugging:\n";
+    var_dump($f);
+    debug_zval_dump($f);
```

Comment:
As pointed out by @hikari-no-yume on the mailing list we need to test the behaviour of `print_r($f)` and
```php
settype($f, 'string); // $f is modified by reference
echo $f;
```
We probably should try to double check which other function are/should be affected by this change.

Reply:
Done! I found some other affected functions as well